### DISCUSSION TITLE
width for spinbox

### DISCUSSION THREAD

**Discussion by azim-b** [h2oai/wave#1082] [Ideas] [answered]
Like dropdown, it'd be nice to have an option to set the width of a `ui.spinbox()`

Since the label's length could be variable, spinboxes within a "form" don't align well.

For example:

![image](https://user-images.githubusercontent.com/64787868/138301465-9faa384b-96fe-4a19-8164-a76f3a7d3595.png)

**Comment by mturoci** [comment, marked as answer]
Hi @azim-b,

`ui.spinbox` currently supports `width` attr, but it allows you to control the whole width of spinbox (label + input). I assume you want to control input only, which is not possible.

I checked and there was a small bug that didn't expand the spinbox when tooltip was specified (your screenshot above):
![image](https://user-images.githubusercontent.com/64769322/138414728-55a04310-0c63-44a2-9260-1126f4fcbe78.png)

Fixed:
![image](https://user-images.githubusercontent.com/64769322/138414539-1720818d-7744-47cf-90a0-084dab7463fe.png)

To further improve this, we could introduce `inline` attr that would render label above instead of inline:
![image](https://user-images.githubusercontent.com/64769322/138415002-3e7b6f6e-3cbb-48de-8eef-c1254c8ec5a9.png)
vs
![image](https://user-images.githubusercontent.com/64769322/138415076-b8bf2c86-bf8f-4f9c-87f8-db682539c7d2.png)

**Reply by azim-b** [reply]
thanks @mturoci - this is helpful!

I didn't know that there was an option to control the whole width of the spin box (label + input); that'd actually solve the problem. And also would the bug fix that expands the spin box input.

Like dropdown, a spinbox where the input is rendered below the label would also be a useful option.

**Reply by mturoci** [reply]
Glad it helped. Also make sure you use tooltips only when really necessary as they are not a good UX (user needs to do additional action instead of understanding what needs to be filled in right away). Prefer clear descriptive labels and use tooltips as an escape hatch only.

**Reply by azim-b** [reply]
gotcha - good suggestion. I'll keep that in mind.
